torch_xla/csrc/tensor.cpp

```diff
@@ -651,7 +651,12 @@ void XLATensor::zero_(XLATensor& input) {
 }
 
 void XLATensor::s_copy_(XLATensor& input, const XLATensor& src) {
-  input.SetIrValue(src.GetIrValue());
+  if (input.GetDevice() == src.GetDevice()) {
+    input.SetIrValue(src.GetIrValue());
+  } else {
+    XLA_ERROR() << "Cannot move from " << input.GetDevice() << " to "
```

Comment:
This is not happening ATM. Will do in followup PR.

Reply:
What would we need this for? I see you went for `DataParallel` taking the training loop, so we shouldn't have to copy anything between devices.

Reply:
To make sure we can move data across devices with to().
Yes, today we do not use it, but it should be do crap like we'd do otherwise :)

Reply:
I see, definitely a good fix, just didn't see how it was related to this PR.